src/BinaryParsers/PlatformSpecificHelpers.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.CodeAnalysis.BinaryParsers
+{
+    public static class PlatformSpecificHelpers
+    {
+        public static bool RunningOnWindows()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+        public static void ThrowIfNotOnWindows()
+        {
+            if(!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                throw new PlatformNotSupportedException(
+                    string.Format(BinaryParsersResources.PlatformUnsupportedFormat, RuntimeInformation.OSDescription, OSPlatform.Windows));
```

Comment:
NIT, note that our source is currently inconsistent in whether we use the C# aliases or formal .NET primitive names when invoking statics on the types. I think CodeFormatter can help clean this up. StyleCop may be able to as well, another option.